livre/core/util/Utilities.h

```diff
@@ -25,6 +25,46 @@ namespace livre
 
 bool hasOpenGL42Support();
 
+/**
+ * Compute the frame range based on the specified range and the boundaries
+ * provided (e.g. data range).
+ * @param frameRange the specified frame range used as a reference to compute
+ * the frame range to return
+ * @param boundaries the range that will be used as a hard limit to compute
+ * the frame number to return
+ * @return a valid frame range, or INVALID_FRAME_RANGE if frameRange is outside
+ * the boundaries
+ */
+Vector2ui computeFrameRange( const Vector2ui& frameRange,
+                             const Vector2ui& boundaries );
+
+/**
+ * Compute the current frame number, based on the value used as input, as well
+ * as the frame range specified. Optionally, it can make the
+ * returned value stick to the latest available
+ * @param frameNumber the value used as input to determine the frame number
+ * @param frameRange the frame range [x, y), open at the end, in which the
+ * returned value must be contained
+ * @param latestAlways if true, return always the last frame number available:
+ * (y - 1)
+ * @return the current frame number
+ */
+uint32_t computeCurrentFrameNumber( uint32_t frameNumber, bool latestAlways,
+                                    const Vector2ui& frameRange );
+
+/**
+ * Compute the next frame number, based on the current frame number and the
+ * delta used as input, as well as the frame range specified.
+ * @param current the current frame number, used as a reference to compute the
+ * next frame number
+ * @param delta the value in which the frame number will be incremented.
+ * It can also be negative
+ * @param frameRange the frame range [x, y), open at the end, in which the
+ * returned value must be contained
+ * @return the next frame number
+ */
+uint32_t computeNextFrameNumber( uint32_t current, int32_t delta,
```

Comment:
The documentation for this function should indicate that it loops back to the beginning when the value exceeds the max frame number